File: Hearthstone-Script/src/main/java/club/xiaojiawei/custom/LogThread.java
package club.xiaojiawei.custom;

import club.xiaojiawei.bean.LogRunnable;

/**
 * @author 肖嘉威
 * @date 2024/9/8 11:14
 */
public class LogThread extends Thread{

    public LogThread(Runnable task) {
        super(new club.xiaojiawei.bean.LogRunnable(task));
    }

    public LogThread(Runnable task, String name) {
        super(new LogRunnable(task), name);
    }
}

File: Hearthstone-Script-Base/src/main/kotlin/club/xiaojiawei/bean/LogThread.kt
package club.xiaojiawei.bean

/**
 * @author 肖嘉威
 * @date 2024/9/8 18:23
 */
open class LogThread : Thread {

    constructor(task: Runnable?) : super(LogRunnable(task))
    constructor(task: Runnable?, name: String?) : super(LogRunnable(task), name)

}